src/ClientResolver.php

```diff
@@ -1044,4 +1066,45 @@ private static function isValidRegion($region)
     {
         return is_valid_hostlabel($region);
     }
+
+    private function _apply_client_context_params(array $args)
+    {
+        if (isset($args['api'])) {
+            if (!is_null($clientContextParams = $args['api']->getClientContextParams())) {
+                foreach($clientContextParams as $paramName => $paramSpec) {
+                    $definition = [
+                        'type' => 'value',
+                        'valid' => [$paramSpec['type']],
+                        'doc' => isset($paramSpec['documentation']) ?
```

Comment:
We look at it like we would in the defaultArgs— the doc is provided in the IAE:

```
Fatal error: Uncaught InvalidArgumentException: Invalid configuration value provided for "UseArnRegion". Expected boolean, but got string(4) "true"

UseArnRegion: (boolean)

  Enables this client to use an ARN's region when constructing an endpoint
  instead of the client's configured region. in /Users/xxx/xxx/xxx/aws-sdk-php/src/ClientResolver.php:417